### PR TITLE
Duplicate docs

### PR DESCRIPTION
**Pull request by cosmingodeanu**

### CODE REVIEW

File: src/main/java/com/code4ro/legalconsultation/model/dto/documentnode/DocumentNodeCreateDto.java [hunk 1]
```diff
@@ -11,7 +13,9 @@
     private UUID parentId;
     private DocumentNodeType documentNodeType;
     @Size(max = 255)
+    @UniqueDocumentNodeTitleConstraint
```
Comment (mihairinzis) [inline src/main/java/com/code4ro/legalconsultation/model/dto/documentnode/DocumentNodeCreateDto.java:16]: optional change: you can drop 'Constraint' from the annotation name. i think the 'Unique' label is quite suggestive already

File: src/test/java/com/code4ro/legalconsultation/controller/DocumentControllerIntegrationTest.java [hunk 2]
```diff
@@ -123,7 +123,30 @@ public void saveDocumentDuplicatedFilePath() throws Exception {
                 .content(Jackson.toJsonString(secondDocument))
                 .accept(MediaType.APPLICATION_JSON))
                 .andExpect(status().isBadRequest());
+    }
 
+    @Test
+    @WithMockUser
+    @Transactional
+    public void saveDocumentDuplicatedTitle() throws Exception {
```
Comment (mihairinzis) [inline src/test/java/com/code4ro/legalconsultation/controller/DocumentControllerIntegrationTest.java:131]: no test for content?

Comment (cosmingodeanu) [reply src/test/java/com/code4ro/legalconsultation/controller/DocumentControllerIntegrationTest.java:131]: should the test for content concern the DocumentNodeController? are you saying I should make tests for the title and content of the DocumentNodeDTO? I couldn't find a content field besides the node dto

Comment (mihairinzis) [reply src/test/java/com/code4ro/legalconsultation/controller/DocumentControllerIntegrationTest.java:131]: yes, i see that you have two validators but only test one. but i actually waonder why do we need a unique content validator

Comment (cosmingodeanu) [reply src/test/java/com/code4ro/legalconsultation/controller/DocumentControllerIntegrationTest.java:131]: According to this https://github.com/code4romania/expert-consultation-api/issues/147  if we don't have one 'there'll be two laws oug with different id's only"

Comment (mihairinzis) [reply src/test/java/com/code4ro/legalconsultation/controller/DocumentControllerIntegrationTest.java:131]: ok so i think both validators are wrong. we should not validate the document node. a document node is a parsed section of a document. we should validate the document creation/update: `/api/documents` DocumentViewDto.documentTitle and DocumentViewDto.filePath. i don't think that checking the whole content makes a lot of sense..

Comment (cosmingodeanu) [reply src/test/java/com/code4ro/legalconsultation/controller/DocumentControllerIntegrationTest.java:131]: But the file path can be changed so even though it's different, the file itself is the same so the content is duplicated.

Comment (mihairinzis) [reply src/test/java/com/code4ro/legalconsultation/controller/DocumentControllerIntegrationTest.java:131]: yes, the best way would probably be to store a pdf hash and store that but i'm not sure we actually need this. opinions @RaduCStefanescu @georgebejan ?

Comment (mihairinzis) [reply src/test/java/com/code4ro/legalconsultation/controller/DocumentControllerIntegrationTest.java:131]: i'd say just keep the  name for now

**Pull request closed without merging**